userspace/libscap/scap.h

```diff
@@ -199,6 +199,7 @@ typedef struct scap_threadinfo
 	uint64_t sid; ///< The session id of the process containing this thread.
 	char comm[SCAP_MAX_PATH_SIZE]; ///< Command name (e.g. "top")
 	char exe[SCAP_MAX_PATH_SIZE]; ///< argv[0] (e.g. "sshd: user@pts/4")
+	char exepath[SCAP_MAX_PATH_SIZE]; ///< full executable path
```

Comment:
I don't think that's what I was talking about, I was more referring to the fact that, should you export the `filename` field in a sysdig filter (like I suggested), then all the processes part of the initial scap state wouldn't have it, and so it would be inconsistent with the processes for which you see an execve. Not only that, but the field would be filled with garbage since it doesn't seem like you are zeroing it in `scap_read_proclist` (my other comment).

I think this behavior you are pointing out is caused by something else.